superset-frontend/plugins/plugin-chart-echarts/src/defaults.ts

```diff
@@ -23,7 +23,7 @@ export const defaultGrid = {
 };
 
 export const defaultTooltip = {
-  confine: true,
+  confine: false,
```

Comment:
IIRC, we originally added this to make sure the tooltip doesn't leave the window. See https://github.com/apache/echarts/issues/5004. However, this does have the drawback of being suboptimal if the chart is small and the tooltip is big. It would be nice if ECharts automatically made sure the tooltip isn't cut, but in the interim, maybe we could disable `confine` as suggested and use a custom position callback to make sure it's within the window's boundaries but doesn't overlap the pointer? 
```
position: function(pt) {
  const x = <some logic based on window width etc>;
  const y = <some logic based on window height etc>;
  return [x, y];
},
```

Check the example here by clicking "Try it" to experiment: https://echarts.apache.org/en/option.html#tooltip.confine

<img width="1318" alt="image" src="https://user-images.githubusercontent.com/33317356/201314242-07a916fc-328f-4e7b-b67c-46ff09511a8f.png">

Reply:
Thanks for the context @villebro. What I noticed is that `confine` is used when overriding the positioning function. I wasn't able to make the tooltip overflow when using the default positioning function with `confine: false`.

https://user-images.githubusercontent.com/70410625/201763499-9ce58788-ef88-460d-8e57-20f871195083.mov

Reply:
Oh ok, I hadn't noticed that. I dug out the original PR that enabled confine: https://github.com/apache-superset/superset-ui/pull/764 Here's a screencapture of way too many items on the tooltip with `confine: true`:

https://user-images.githubusercontent.com/33317356/202183731-fa6e8310-5f25-4853-83a0-b30c2991c58a.mp4

And the same with `confine: false`:

https://user-images.githubusercontent.com/33317356/202183890-304cf230-82fb-4947-a793-7134b82131da.mp4

So I believe one unwanted side-effect of disabling `confine` is that the tooltip position happily overflows the top of the screen. Looking at the docs, I tested the example position they gave, and cleverly using those additional objects that are passed to the callback, it should be possible to do some exotic positioning that is as convenient as possible:

```javascript
{
  ...,
  position: function (pos, params, dom, rect, size) {
      // tooltip will be fixed on the right if mouse hovering on the left,
      // and on the left if hovering on the right.
      var obj = {top: 60};
      obj[['left', 'right'][+(pos[0] < size.viewSize[0] / 2)]] = 5;
      return obj;
  },
}
```
https://user-images.githubusercontent.com/33317356/202188068-04a8ce49-279c-43a9-9efb-0479ed16ea4e.mp4

https://echarts.apache.org/en/option.html#tooltip.position